src/graphnet/data/i3extractor.py

```diff
@@ -304,7 +304,29 @@ def __call__(self, frame, padding_value=-1) -> dict:
                     'final_position_y': final_position[1],
                     'final_position_z': final_position[2],
                     'stopped_muon': stopped,
-                })                
+                })  
+            if frame_contains_retro(frame):
+                if abs(MCInIcePrimary.pdg_encoding) != 13: # if not muon
+                    output.update({
+                        'osc_weight'         : frame["I3MCWeightDict"]["weight"],
+                        'OneWeight'          : frame["I3MCWeightDict"]["OneWeight"],
+                        'gen_ratio'          : frame["I3MCWeightDict"]["gen_ratio"],
+                        'NEvents'            : frame["I3MCWeightDict"]["NEvents"],
```

Comment:
@RasmusOrsoe, are these weight attributes specific to RetroReco, or OscNext?

Reply:
These quantities have nothing to do with Retro and are variables that comes from the simulation engine.

Reply:
In that case, is there a reason we enclose them in the `if frame_contains_retro(frame):` block? If these are from Genie (or whatever), should we check for this instead?

Reply:
They are currently enclosed in this check because I know from experience `'osc_weight'` is not always available, but is always available when Retro is applied. I assume its the same for these keys.

I can remove the frame check but I would then instead have to wrap every single quantity into a

```
try:
    output['osc_weight'] = frame["I3MCWeightDict"]["weight"]
except:
    output['osc_weight']  = padding_value
```

Reply:
We could add a utility method like
```python
from typing import Any, Union

def try_get_key(frame: Union[I3Frame, dict], key: str, default_value: Any) -> Any:
    """Return `key` in `frame` if it exists; otherwise return `default_value."""
    try:
        return frame[key]
    except KeyError:
        return default_value
```
or, even better, if we know that `frame` and all of it's "child" objects are python-`dict`-like, we can just use the built-in method `.get(key, default_value)` (see https://docs.python.org/3/library/stdtypes.html#dict.get), i.e.
```python
output['osc_weight'] = try_get_key(frame["I3MCWeightDict"], "weight", padding_value)

# or, even better, if frame["I3MCWeightDict"] is dict-like
output['osc_weight'] = frame["I3MCWeightDict"].get("weight", padding_value)
```